### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by FevenSeyfu**
In this PR I have completed the below requirements:

- Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as your backend.

- Use [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make your HTTP requests to the backend

- Fetch your books from the Bookstore API after the component has rendered

- Refactor your add book and remove book features to persist your changes in the server

- Your application should behave the same way after the refactoring

### CODE REVIEW

File: src/components/Books.js [hunk 1]
```diff
@@ -1,20 +1,20 @@
-import { useSelector } from 'react-redux';
+// import { useSelector } from 'react-redux';
```
Comment (MahmoudBakr23) [inline src/components/Books.js]: Required
- Please remove the old pieces of code for a cleaner code base.

File: src/components/Books.js [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 import BookCard from './BookCard';
 import Form from './AddNewBook';
 
 function Books() {
-  const { booksList } = useSelector((state) => state.book);
   return (
     <section className="ListBooks">
-      {booksList.map((book) => (
+      {/* {booksList.map((book) => (
```
Comment (MahmoudBakr23) [inline src/components/Books.js]: Required
- Please remove the old pieces of code for a cleaner code base.

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @FevenSeyfu 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up!
- Great job creating the async functions
- Well done refactoring the add and remove functions
- Nice work using axios and fetch
- Descriptive pr and readme file
- Correct Linters 

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Books.js (hunk 2 above)
Comment (FevenSeyfu) [reply src/components/Books.js]: thank you for the feedback. I have removed the old pieces of code.

Comment (descholar-ceo) [reply src/components/Books.js]: 👍🏽 Addressed ✔️

File: src/components/Books.js (hunk 1 above)
Comment (descholar-ceo) [reply src/components/Books.js]: 👍🏽 Addressed ✔️

**Review by descholar-ceo — APPROVED** [review]
Hi @FevenSeyfu 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Addressed all comments from the previous previous review ✔️ 
👍🏽 Linters pass the check ✔️ 
👍🏽 Commit message and the readme file are professional ✔️ 
👍🏽 Correct usage of the git/github flow ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**